Source/Core/VideoCommon/TextureDecoder_Common.cpp

```diff
@@ -376,35 +376,35 @@ static inline u32 DecodePixel_IA8(u16 val)
 {
   int a = val & 0xFF;
   int i = val >> 8;
-  return i | (i << 8) | (i << 16) | (a << 24);
+  return (i * 0x010101) | (a << 24);
```

Comment:
Again the profiling question, but is this noticable enough to sacrifice the readability of how `i` contributes to the result? Same for the other `* 0x010101` too.
If it does make a significant difference, I'd like a comment that explains the construction of the resulting value from the given parameters (similar to the swizzling comments perhaps? or at least the old "readable" code snippet) so others can understand whats going on.